src/validators/OidcValidator.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import { Transaction } from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/TransactionHelper.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import { IModuleValidator } from "../interfaces/IModuleValidator.sol";
+import { IModule } from "../interfaces/IModule.sol";
+import { VerifierCaller } from "../helpers/VerifierCaller.sol";
+
+/// @title OidcValidator
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @dev This contract allows secure user authentication using OIDC protocol.
+contract OidcValidator is VerifierCaller, IModuleValidator {
+  event OidcKeyUpdated(address indexed account, bytes iss, bool isNew);
+
+  struct OidcData {
+    bytes oidcDigest; // PoseidonHash(sub || aud || iss || salt)
+    bytes iss; // Issuer
+    bytes aud; // Audience
+  }
+
+  struct OidcSignature {
+    bytes zkProof;
+    bytes kid; // Key id used in the jwt
+  }
+
+  mapping(address => OidcData) public accountData;
+
+  /// @notice Runs on module install
+  /// @param data ABI-encoded OidcData key to add immediately, or empty if not needed
+  function onInstall(bytes calldata data) external override {
+    if (data.length > 0) {
+      require(addValidationKey(data), "OidcValidator: key already exists");
+    }
+  }
+
+  /// @notice Runs on module uninstall
+  /// @param data unused
+  function onUninstall(bytes calldata data) external override {
+    accountData[msg.sender] = OidcData(bytes(""), bytes(""), bytes(""));
+  }
+
+  /// @notice Adds an `OidcData` for the caller.
+  /// @param key ABI-encoded `OidcData`.
+  /// @return true if the key was added, false if it was updated.
+  function addValidationKey(bytes calldata key) public returns (bool) {
```

Comment:
Consider adding tests while you write this code. I believe that it's going to make you advance stronger :)